### PR TITLE
immer wallets 2

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] makeReducer / immer of root

### CODE REVIEW

**Comment by chrisnojima** [comment]
cc: @keybase/picnicsquad

**Comment by cjb** [comment]
Will the ts-ignores go away once the later branches are merged down? Maybe I should wait to approve this PR until then..

File: shared/reducers/wallets.tsx [hunk 1]
```diff
@@ -26,618 +26,595 @@ const reduceAssetMap = (
...
+    )
+    draftState.accountMap = accountMap
+  },
+  [WalletsGen.changedAccountName]: (draftState, action) => {
+    const {account} = action.payload
+    // accept the updated account if we've loaded it already
+    // this is because we get the sort order from the full accounts load,
+    // and can't figure it out from these notifications alone.
+    if (account) {
+      // } && state.accountMap.get(account.accountID)) {
```
Comment (songgao) [inline shared/reducers/wallets.tsx]: 🔥

File: shared/reducers/wallets.tsx [hunk 2]
```diff
@@ -26,618 +26,595 @@ const reduceAssetMap = (
...
+    Object.keys(o).forEach(k => {
+      const so = JSON.stringify(o[k])
+      const sn = JSON.stringify(n[k])
+      if (so !== sn) {
+        same = false
+        console.log('aaa diff', k, so, sn, action)
+      }
+    })
+    if (same) {
+      console.log('aaa same')
```
Comment (songgao) [inline shared/reducers/wallets.tsx]: temp logging? I haven't looked at the rest of these so maybe it's already removed.

**Pull request merged**